main.go

```diff
@@ -142,7 +144,32 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 	if resp.StatusCode != http.StatusOK {
 		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
 	}
+	contentCheck(resp, ch)
+}
 
+func fetchMetricFamiliesWithAuth(url string, ch chan<- *dto.MetricFamily, certificate string, key string) {
+	defer close(ch)
+	cert, err := tls.LoadX509KeyPair(certificate, key)
+	if err != nil {
+		log.Fatal(err)
+	}
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	}
+	tlsConfig.BuildNameToCertificate()
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client := &http.Client{Transport: transport}
+	resp, err := client.Get(url)
```

Comment:
You need to set the accept header here, too.

In general, I would factor out only the creation of the client (TLS or not), and then the rest of the code can be shared.